### PR TITLE
feat(proxy): fix TikTok Ads access_token header for proxy requests

### PR DESCRIPTION
**Pull request by hassan254-prog**
## Describe your changes

- Dynamically build authentication in headers for TikTok Ads. 

## Issue ticket number and link
[EXT-167](https://linear.app/nango/issue/EXT-167/tiktok-ads-access-token-should-be-in-the-access-token-header-for-proxy)

### CODE REVIEW

File: packages/types/lib/providers/provider.ts [hunk 1]
```diff
@@ -54,6 +54,10 @@ export interface BaseProvider {
             base_url_override?: string;
             headers?: Record<string, string>;
         };
+        authentication_header?: {
+            key: string;
+            value: string;
+        };
```
Comment (bodinsamuel) [inline packages/types/lib/providers/provider.ts]: What if you need more than 1 header?
Also the format is different than `headers` which seems to accommodate more than one header, might be a good idea to follow the same pattern

**Comment by hassan254-prog** [comment]
I will focus on fixing headers for TikTok for this specific task  and open another PR when we may have a new `JWT` provider with dynamic headers.

**Review by khaliqgant — APPROVED** [review]

**Comment by bodinsamuel** [comment]
@hassan254-prog the PR seems to be very different now, is the title still correct or not ready to review?

**Comment by hassan254-prog** [comment]
> @hassan254-prog the PR seems to be very different now, is the title still correct or not ready to review?

Yes, I’ve renamed the title to reflect the new changes, sorry about that.

**Review by bodinsamuel — APPROVED** [review]

**Pull request merged**